src/state/selection/selectors/selectors.test.ts

```diff
@@ -35,50 +39,137 @@ const mockState: State = {
+            expect(highlightedNames).toEqual([{ name: "agent3", tags: [] }]);
+        });
+        it("returns an empty array when the agent has no display states and no tags", () => {
+            const stateWithSelection = {
+                ...mockState,
+                selection: {
+                    ...mockState.selection,
+                    highlightedAgentsNamesAndTags: {
+                        agent3: [],
+                    },
+                },
+            };
+            const highlightedNames = getHighLightedAgents(stateWithSelection);
+            expect(highlightedNames).toBeInstanceOf(Array);
+            expect(highlightedNames).toEqual([]);
+        });
+        it("returns an array of agent names and tags, including an unmodified tag if present", () => {
+            const stateWithSelection = {
+                ...mockState,
+                selection: {
+                    ...mockState.selection,
+                    highlightedAgentsNamesAndTags: {
+                        agent2: ["", "state1"],
+                    },
+                },
+            };
+            const highlightedNames = getHighLightedAgents(stateWithSelection);
+            expect(highlightedNames).toBeInstanceOf(Array);
+            expect(highlightedNames).toEqual([
+                { name: "agent2", tags: ["state1", ""] },
+            ]);
+        });
         it("only returns names included in the display data from the backend", () => {
             const stateWithSelection = {
                 ...mockState,
                 selection: {
                     ...mockState.selection,
-                    highlightedAgentKeys: { agent1: ["blah"] },
+                    highlightedAgentsNamesAndTags: { agent1: ["blah"] },
                 },
             };
-            const highlightedNames = getHightLightedAgents(stateWithSelection);
+            const highlightedNames = getHighLightedAgents(stateWithSelection);
             expect(highlightedNames).toBeInstanceOf(Array);
             expect(highlightedNames).toEqual([]);
         });
     });
-    describe("getAgentNamesToHide", () => {
+    describe("getAgentsToHide", () => {
         it("returns an array of agent names that are not currently selected", () => {
             const stateWithSelection = {
                 ...mockState,
                 selection: {
                     ...mockState.selection,
-                    visibleAgentKeys: { agent1: ["agent1", "v-agent2-state2"] },
+                    visibleAgentsNamesAndTags: {
+                        agent1: ["agent1", "v-agent2-state2"],
+                        agent2: [""],
+                    },
                 },
             };
             const hiddenNames = getAgentsToHide(stateWithSelection);
             expect(hiddenNames).toBeInstanceOf(Array);
             expect(hiddenNames).toEqual([
                 { name: "agent1", tags: ["state1", ""] },
+                { name: "agent2", tags: ["state1", "state2"] },
```

Comment:
@meganrm I'm actually still kinda confused about this function `getAgentsToHide`. Why is it written so that if there are 3 agents, and only 2 agents are under `visibleAgentsNamesAndTags`, the third agent is not included in the list of hidden names? i.e. if `agent3` is not listed under `visibleAgentsNamesAndTags` shouldn't it be listed under `hiddenNames`? I see that `getAgentsToHide` is written to keep `agent3` from being added to the output, but why do we want/need it that way?

Reply:
Just talked with Megan and renamed some more variables and rewrote these tests